tensorflow/compiler/xla/service/hlo_dataflow_analysis.cc

```diff
@@ -120,6 +120,139 @@ bool HloDataflowAnalysis::AreTransitiveUsesElementwiseOrTuple(
   return true;
 }
 
+namespace {
+bool Is1dSliceWithoutStrides(const HloInstruction* instr) {
+  return instr->opcode() == HloOpcode::kSlice &&
+         1 == instr->slice_starts().size() &&
+         1 == instr->slice_limits().size() &&
+         1 == instr->slice_strides().size() &&
+         1 == instr->slice_strides().at(0);
+}
+
+bool IsSliceInputFusion(const HloInstruction& unnested_hlo) {
```

Comment:
What do you think about rewriting this function using HLO matchers? (cf. pattern_matcher.h and the files which use it), it seems like it could be a lot smaller and a lot more readable?